src/fides/api/models/privacy_request.py

```diff
@@ -668,15 +684,44 @@ def get_manual_webhook_input_strict(
             f"No data cached for privacy_request_id '{self.id}' for connection config '{manual_webhook.connection_config.key}'"
         )
 
-    def get_manual_webhook_input_non_strict(
+    def get_manual_webhook_erasure_input_strict(
+        self, manual_webhook: AccessManualWebhook
+    ) -> Dict[str, Any]:
+        """
+        Retrieves manual webhook fields saved to the privacy request in strict mode.
+        Fails either if extra saved fields are detected (webhook definition had fields removed) or fields were not
+        explicitly set (webhook definition had fields added). This mode lets us know if webhooks data needs to be re-uploaded.
+
+        This is for use by the *manual_webhook* connector which is *NOT* integrated with the graph.
+        """
+        cached_results: Optional[Dict[str, Any]] = _get_manual_erasure_input_from_cache(
+            privacy_request=self, manual_webhook=manual_webhook
+        )
+
+        if cached_results:
+            data: Dict[str, Any] = manual_webhook.erasure_fields_schema.parse_obj(
+                cached_results
+            ).dict(exclude_unset=True)
+            if set(data.keys()) != set(
```

Comment:
i could be following wrong, but is there a narrow (and perhaps too contrived) edge case where the value has been set to `False` for a given field (i.e. someone actually checks and unchecks a field) and we essentially judge this as OK, we're ready to proceed, when in reality they still have one or more fields outstanding? maybe that's a desired behavior, i.e. if an admin has explicitly unchecked a box then its up to their discretion to allow the request to proceed...

anyway, sorta just curious whether that's a legitimate possibility and if so whethere you'd thought thru it at all...

Reply:
I think this is ok, but that's only because it's hard for us to determine when it's acceptable to proceed with a manual erasure. There are cases when certain fields might not exist in the manual data source so we'll still want to proceed even if no fields are checked.

Reply:
OK fair enough, i mean i think there's a bit of tension between that and what we're saying [here](https://github.com/ethyca/fides/pull/4115#discussion_r1337649410), i.e. that completed deletion infers a confirmation was made. if we're allowing deletion to proceed without requiring confirmation of all fields, someone who is auditing things really has no way of determining what was deleted on a particular DSR request.

totally acknowledge that this is an edge case and the requirement generally is perhaps beyond MVP scope. i just think that a robust/long term implementation probably includes some sort of persisted tracking of what was confirmed on a particular DSR 👍